### PR TITLE
Introduce `orchard.inspect/nav-to-item`

### LINKED ISSUE #209: Inspector: add op to nav a sub item
Unlike e.g. Portal, we don't have a command to `nav` to a sub-item (e.g. nth, or k-v entry) and re-render the Inspector using the produced value.

### PR DESCRIPTION
**Pull request by vemv**
Closes https://github.com/clojure-emacs/orchard/issues/209

### CODE REVIEW

File: src/orchard/inspect.clj [hunk 1]
```diff
@@ -508,19 +539,25 @@
         (unindent))
     inspector))
 
-(defn- nav-datafy-tx [obj remove-nil-valued-entries?]
+(defn- nav-datafy-tx [remove-nil-valued-entries?]
   (keep (fn [[k v]]
-          (or (some->> (nav obj k v)
```
Comment (vemv) [inline src/orchard/inspect.clj:513]: Note the deletion of `nav`, explained in the comment below.

I'd say it was a bug - deftests were showing that we were performing `nav` over the same items repeatedly, which can be expensive in the face of side-effects.

File: test/orchard/inspect_test.clj [hunk 2]
```diff
@@ -941,32 +958,6 @@
                         (:newline))
                       (datafy-section rendered))))))))
 
-(deftest inspect-navigable-metadata-extension-test
```
Comment (vemv) [inline test/orchard/inspect_test.clj:944]: Test deleted as it was based on eagerly calling `nav`.

The added test already shows the behavior for when calling nav intentionally.

**Comment by vemv** [comment]
Hmm, the basic idea works but this will need some refinements since the `index` has some edge cases.
